src/number_theory/divisors.lean

```diff
@@ -202,4 +201,39 @@ begin
   { apply add_right_cancel h }
 end
 
+lemma mem_divisors_prime_pow {p : ℕ} (pp : p.prime) (k : ℕ) {x : ℕ} :
+  x ∈ divisors (p ^ k) ↔  ∃ (j : ℕ) (H : j ≤ k), x = p ^ j :=
+by rw [mem_divisors, nat.dvd_prime_pow pp, and_iff_left (ne_of_gt (pow_pos pp.pos k))]
+
+lemma divisors_prime {p : ℕ} (pp : p.prime) :
+  divisors p = {1 , p} :=
+begin
+  ext,
+  simp only [pp.ne_zero, and_true, ne.def, not_false_iff, finset.mem_insert,
+    finset.mem_singleton, mem_divisors],
+  refine ⟨pp.2 a, λ h, _⟩,
+  rcases h; subst h,
+  apply one_dvd,
+end
+
+lemma divisors_prime_pow {p : ℕ} (pp : p.prime) (k : ℕ) :
+  divisors (p ^ k) = (finset.range (k + 1)).map ⟨pow p, pow_right_injective pp.two_le⟩ :=
+by { ext, simp [mem_divisors_prime_pow, pp, nat.lt_succ_iff, @eq_comm _ a] }
+
+open finset
+@[simp]
+lemma sum_divisors_prime {α : Type*} [add_comm_monoid α] {p : ℕ} {f : ℕ → α} (h : p.prime) :
```

Comment:
Do you want to write this as `prod_divisors_prime` and use `to_additive` to get this version?

Reply:
The fact that there's a 1 in the statement appears to be confusing `to_additive`. I'm not sure how to fix that.

Reply:
Ah fair enough - you could make `prod_divisors_prime` and use `sum_divisors_prime := @prod_divisors_prime (multiplicative α) ...` in that case, but it's up to you.

Reply:
Thanks for the tip!